src/HealthGPS/energy_balance_hierarchical_model.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include "hierarchical_model_types.h"
+#include "random_algorithm.h"
+
+namespace hgps {
+
+	/// @brief Implements the dynamic hierarchical linear model (energy balance) type
+	/// 
+	/// @details The dynamic model is used to advance the virtual population over time.
+	class EnergyBalanceHierarchicalModel final : public HierarchicalLinearModel {
+	public:
+		EnergyBalanceHierarchicalModel() = delete;
```

Comment:
I don't *think* you need this.

Reply:
Any reason not to delete default constructors in stateful classes in general?